todo-list/fn_and_obj.py

```diff
@@ -0,0 +1,129 @@
+        todo_list = pickle.load(infile)
+        infile.close()
+        use_old_list = input("Previous list found. Do you want to load? (Y/N) ")
+        if use_old_list[0].lower() == 'y':
+            return todo_list
+        else:
+            return ToDoList()
+        
+    except:
+        print('No previous list found. Staring with empty to do list.')
+        return ToDoList()
+    
+
+# Create class for list of ToDoItem, with required functionality
+class ToDoList:
+    
+    def __init__(self):
+        self.actual_list = []
+        
+    def add_item(self, item):
+         self.actual_list.append(item)
+        
+    def remove_item(self, item_r):
+        for i in range(len(self.actual_list)):
+            if self.actual_list[i].description == item_r:
+                del self.actual_list[i]
+                break
+        print('Item does not exist')
+                
+    # This function can inform whether an ToDoItem can be editted or if it doesn't exist
+    def item_exist(self, item_e):
+        for i in range(len(self.actual_list)):
+            if self.actual_list[i].description == item_e:
+                return True
+        return False 
+    
+    # Finding the index of a ToDoItem allows us to then call any item method.
+    # Otherwise each ToDoItem method would require a separate ToDoList method.
+    def find_item(self, item_f):
+        for i in range(len(self.actual_list)):
+            if self.actual_list[i].description == item_f:
+                return i
+        
+    # Each print option has a separate method (not condensed due to different printing)
+    def print_items(self):
+        for i in range(len(self.actual_list)):
+            if self.actual_list[i].complete:
+                completed_str = ' is completed'
+            else:
+                completed_str = ' is not completed'
+            print(self.actual_list[i].description + ' due on ' + 
+                      self.actual_list[i].date_str + completed_str)
+            
+    def print_incomp_items(self):
+        for i in range(len(self.actual_list)):
+            if not self.actual_list[i].complete:
+                print(self.actual_list[i].description + ' due on ' + 
+                      self.actual_list[i].date_str)
+    
+    def print_comp_items(self):
+        for i in range(len(self.actual_list)):
+            if self.actual_list[i].complete:
+                print(self.actual_list[i].description + ' due on ' + 
+                      self.actual_list[i].date_str)
+
+    def print_overdue_items(self):
+        for i in range(len(self.actual_list)):
+            if (not self.actual_list[i].complete) and datetime.now() > self.actual_list[i].date_obj:
+                print(self.actual_list[i].description + ' due on ' + 
+                      self.actual_list[i].date_str)
+
+    def print_search_items(self, search):
+        for i in range(len(self.actual_list)):
+            if search.lower() in self.actual_list[i].description.lower():
```

Comment:
Again, could have maybe added a `match()` method to your TodoItem, that would return whether a particular string matched, that would allow you to change the implementation more easily later and re-use across different areas.